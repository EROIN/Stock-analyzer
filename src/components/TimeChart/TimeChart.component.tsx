import {
  LineChart,
  Line,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  Legend,
  ResponsiveContainer,
} from 'recharts';

import './TimeChart.scss';
// import {TIME_SERIES_DATA} from '../../__mocks__/timeSeries.mocks';
import {
  getChartDataFromAPIData,
  getXAxisFormat,
} from '../../utils/timeSeries.utils';
import {TimeChartProps} from './TimeChart.types';

export const TimeChart = (props: TimeChartProps) => {
  const {chartData} = props;

  if (!chartData) return null;

  const data = getChartDataFromAPIData(chartData);
  return (
    <ResponsiveContainer width="100%" height="100%">
      <LineChart
        width={500}
        height={300}
        data={data}
        margin={{
          top: 5,
          right: 30,
          left: 20,
          bottom: 5,
        }}
      >
        <CartesianGrid strokeDasharray="3 3" />
        <XAxis dataKey="date" tickFormatter={getXAxisFormat} angle={-45} />
        <YAxis />
        <Tooltip />
        <Legend />
        <Line
          type="monotone"
          dataKey="close"
          stroke="#8884d8"
          activeDot={{r: 8}}
        />
        <Line type="monotone" dataKey="high" stroke="#82ca9d" />
      </LineChart>
    </ResponsiveContainer>
  );
};
