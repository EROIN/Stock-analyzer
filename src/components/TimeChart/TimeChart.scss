.search-tab-chart-container {
  width: 400px;
  height: 200px;
  align-self: center;
}

@media only screen and (max-width: 768px) {
  .search-tab-chart-container {
    width: 100%;
    margin: 2em 0;
  }
}
