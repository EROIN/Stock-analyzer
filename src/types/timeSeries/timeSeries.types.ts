export interface TimeSeriesMetaData {
  string: Record<string, string>;
}

export interface TimeSeries {
  '1. open': string;
  '2. high': string;
  '3. low': string;
  '4. close': string;
  '5. volume': string;
}

export interface TimeSeriesDaily {
  string: TimeSeries;
}

export interface GetTimeSeriesAPIResponse {
  'Meta Data': TimeSeriesMetaData;
  'Time Series (Daily)': TimeSeriesDaily;
}
