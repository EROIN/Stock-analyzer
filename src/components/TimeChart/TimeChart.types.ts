import {GetTimeSeriesAPIResponse} from '../../types/timeSeries/timeSeries.types';

export interface TimeChartProps {
  chartData?: GetTimeSeriesAPIResponse;
}
