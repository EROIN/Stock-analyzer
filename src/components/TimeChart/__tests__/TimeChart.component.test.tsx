import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import {TimeChart} from '../TimeChart.component';
// eslint-disable-next-line jest/no-mocks-import
import {TIME_SERIES_DATA} from '../../../__mocks__/timeSeries.mocks';

describe('<TimeChart/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer
      .create(<TimeChart chartData={TIME_SERIES_DATA} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with empty data', () => {
    const tree = renderer
      //@ts-ignore
      .create(<TimeChart data={{}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
