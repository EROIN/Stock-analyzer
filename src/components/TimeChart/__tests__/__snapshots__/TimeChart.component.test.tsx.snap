// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeChart/> renders correctly with default props 1`] = `
<div
  className="search-tab-chart-container"
>
  <div
    className="recharts-responsive-container"
    style={
      Object {
        "height": "100%",
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`<TimeChart/> renders correctly with empty data 1`] = `null`;
